README.md

```diff
@@ -8,77 +8,65 @@ npm install hypercore-protocol
+const remoteFeed = streamB.open(key, {
+  // listen for data in remote feed
+  ondata (message) {
+    console.log(message.value.toString())
+  }
+})
 
 // add data to feed
 feed.data({ index: 1, value: '{ block: 42 }'})
 
-// listen data in remoteFeed
-remoteFeed.on('data', function (message) {
-  console.log(message.value.toString())
-})
-
 streamA.pipe(streamB).pipe(streamA)
 ```
 
 `output => { block: 42 }`
 
 ## API
 
-#### `var stream = protocol([options])`
+#### `const stream = new Protocol(initiator, [options])`
 
 Create a new protocol duplex stream.
 
 Options include:
 
 ``` js
 {
-  id: optionalPeerId, // you can use this to detect if you connect to yourself
-  live: keepStreamOpen, // signal to the other peer that you want to keep this stream open forever
-  ack: false, // Explicitly ask a peer to acknowledge each received block
-  userData: opaqueUserData // include user data that you can retrieve on handshake
   encrypt: true, // set to false to disable encryption if you are already piping through a encrypted stream
   timeout: 5000, // stream timeout. set to 0 or false to disable.
-  extensions: [], // names of extensions to use for replication. Must be sorted alphanumerically for handshaking to work
+  keyPair: { publicKey, secretKey }, // use this keypair for the stream authentication
+  onhandshake () { }, // function called when the stream handshake has finished
+  onopen (discoveryKey) { } // function called when the remote stream opens a feed you have not
 }
 ```
 
-If you don't specify a peer id a random 32 byte will be used.
-You can access the peer id using `p.id` and the remote peer id using `p.remoteId`.
-
-#### `var feed = stream.feed(key)`
+#### `stream.remotePublicKey`
 
-Signal the other end that you want to share a hypercore feed.
+The remotes public key.
 
-You can use the same stream to share more than one BUT the first feed shared
-should be the same one. The key of the first feed is also used to encrypt the stream using [libsodium](https://github.com/mafintosh/sodium-native#crypto_stream_xorcipher-message-nonce-key).
+#### `stream.publicKey`
 
-#### `var bool = stream.has(key)`
+Your public key.
 
-Returns true if the stream already has open a channel open for the given key and false if not.
+#### `const feed = stream.open(key, handlers)`
 
-#### `stream.on('handshake')`
-
-Emitted when a protocol handshake has been received. Afterwards you can check `.remoteId` to get the remote peer id, `.remoteLive` to get its live status, or `.remoteUserData` to get its user data.
-
-#### `stream.on('feed', discoveryKey)`
-
-Emitted when a remote is sharing a feed. `discoveryKey` is the hypercore discovery key of the feed they want to share.
-
-If you are sharing multiple hypercores on the same port you can use this event to wait for the remote peer to indicate which hypercore
-they are interested in.
+Signal the other end that you want to share a hypercore feed.
+Also sends a capability that you have the specified key.
```

Comment:
I might call this a "proof" or "capability proof" instead of just a "capability." I might also clarify the section like this:

```
Signal the other end that you want to share a hypercore feed.
The feed key will be hashed and sent as the "discovery key" which protects the feed key from being learned by a remote peer who does not already possess it.
Also includes a proof that the local possesses the feed key to protect against replay attacks.
```